backend/eventloop/application_event_loop/application_event_runner_test.go

```diff
@@ -942,6 +942,319 @@ var _ = Describe("ApplicationEventLoop Test", func() {
+			Expect(err).To(BeNil())
+
+			By("Retrieve latest version of GitOpsDeployment and check Health/Sync after calling applicationEventRunner_handleUpdateDeploymentStatusTick function.")
+			clientErr = a.workspaceClient.Get(ctx, gitopsDeploymentKey, gitopsDeployment)
+			Expect(clientErr).To(BeNil())
+
+			Expect(gitopsDeployment.Status.Health.Status).To(Equal(managedgitopsv1alpha1.HeathStatusCodeHealthy))
+			Expect(gitopsDeployment.Status.Sync.Status).To(Equal(managedgitopsv1alpha1.SyncStatusCodeSynced))
+			Expect(gitopsDeployment.Status.Sync.Revision).To(Equal("abcdefg"))
+			Expect(gitopsDeployment.Status.Health.Message).To(Equal("Success"))
+			Expect(gitopsDeployment.Status.ReconciledState.Source.Path).To(Equal(reconciledobj.Source.Path))
+			Expect(gitopsDeployment.Status.ReconciledState.Source.RepoURL).To(Equal(reconciledobj.Source.RepoURL))
+			Expect(gitopsDeployment.Status.ReconciledState.Source.Branch).To(Equal(reconciledobj.Source.TargetRevision))
+			Expect(gitopsDeployment.Status.ReconciledState.Destination.Namespace).To(Equal(reconciledobj.Destination.Namespace))
+
+			matchingCondition, _ := conditions.NewConditionManager().FindCondition(&gitopsDeployment.Status.Conditions, managedgitopsv1alpha1.GitOpsDeploymentConditionInvalidSpecError)
+			Expect(matchingCondition).ToNot(BeNil())
+			Expect(matchingCondition.Message).To(Equal("spec.source.path is a required field and it cannot be empty"))
+			Expect(matchingCondition.Status).To(Equal(managedgitopsv1alpha1.GitOpsConditionStatusTrue))
+			Expect(matchingCondition.Type).To(Equal(managedgitopsv1alpha1.GitOpsDeploymentConditionInvalidSpecError))
+
+			clientErr = a.workspaceClient.Get(ctx, gitopsDeploymentKey, gitopsDeployment)
+			Expect(clientErr).To(BeNil())
+
+			By("Update the path to verify whether gitopsDeployment condition status is false")
+			gitopsDeployment.Spec.Source.Path = "resources/test-data/sample-gitops-repository/environments/overlays/dev"
+			err = k8sClient.Update(ctx, gitopsDeployment)
+			Expect(err).To(BeNil())
+
+			err = a.applicationEventRunner_handleUpdateDeploymentStatusTick(ctx, gitopsDepl.Name, gitopsDepl.Namespace, dbQueries)
+			Expect(err).To(BeNil())
+
+			clientErr = a.workspaceClient.Get(ctx, gitopsDeploymentKey, gitopsDeployment)
+			Expect(clientErr).To(BeNil())
+
+			By("Verify whether gitOpsCondition status is false")
+			matchingCondition, _ = conditions.NewConditionManager().FindCondition(&gitopsDeployment.Status.Conditions, managedgitopsv1alpha1.GitOpsDeploymentConditionInvalidSpecError)
+			Expect(matchingCondition).ToNot(BeNil())
+			Expect(matchingCondition.Status).To(Equal(managedgitopsv1alpha1.GitOpsConditionStatusFalse))
+			Expect(matchingCondition.Message).To(Equal(""))
+
+			By("Delete GitOpsDepl to clean resources.")
+			err = k8sClient.Delete(ctx, gitopsDepl)
+			Expect(err).To(BeNil())
+
+		})
+
+		It("Should update status condition of deployment when path field of gitopsDeployment is '/' ", func() {
```

Comment:
I'm thinking if it's possible to avoid repetition and have a common setup for both tests. WDYT?